trin-core/src/portalnet/types/metric.rs

```diff
@@ -1,22 +1,73 @@
+use std::fmt;
```

Comment:
The filename  `metric.rs` is a little confusing for me, by default, I'm assuming it is something related to Prometheus metrics, maybe change this filename to `distance.rs` or something more appropriate?